drivers/pwm/pwm_mcux_sctimer.c

```diff
@@ -72,42 +120,84 @@ static int mcux_sctimer_pwm_set_cycles(const struct device *dev,
 			base->OUTPUT |= (1UL << channel);
 		}
 
+		/* Set duty cycle to 0 for this channel */
+		SCTIMER_UpdatePwmDutycycle(base, channel, duty_cycle,
+					   data->event_number[channel]);
+		if (data->configured_chan > 1) {
+			/* Make sure the global counter gets restarted */
+			SCTIMER_StartTimer(base, kSCTIMER_Counter_U);
+		} else {
+			/* Only one channel setup, we can safely stop counter.
+			 * we must call this again because the duty cycle
+			 * update function will have started it
+			 */
+			SCTIMER_StopTimer(base, kSCTIMER_Counter_U);
+		}
+
 		return 0;
 	}
 
+	/* SCTimer has some unique restrictions when operation as a PWM output.
+	 * The peripheral is based around a single counter, with a block of
+	 * match registers that can trigger corresponding events. When used
+	 * as a PWM peripheral, MCUX SDK sets up the SCTimer as follows:
+	 * - one match register is used to set PWM output high, and reset
+	 *   SCtimer counter. This sets the PWM period
+	 * - one match register is used to set PWM output low. This sets the
+	 *   pulse length
+	 *
+	 * This means that when configured, multiple channels must have the
+	 * same PWM period, since they all share the same SCTimer counter.
```

Comment:
if this is the case, then why do we have an array of period_cycles for channels

Reply:
You're right- I think this was present for legacy reasons. I'll update the PR to remove it